File: fsharp/BronKerbosch/BronKerbosch2GP.fs
// Bron-Kerbosch algorithm with pivot of highest degree (IK_GP)
module BronKerbosch2GP

open BronKerbosch

let public explore (graph: UndirectedGraph) (consumer: CliqueConsumer) : Unit =
    PivotBased.explore MaxDegreeLocal graph consumer

let algorithm: Algorithm = { name = "Ver2½-GP"; exec = explore }

File: kotlin/src/main/be.steinsomers.bron_kerbosch/BronKerbosch2gp.kt
package be.steinsomers.bron_kerbosch

class BronKerbosch2gp : BronKerboschAlgorithm {
    override val name: String = "Ver2½-GP"

    override fun explore(graph: UndirectedGraph, cliqueConsumer: CliqueConsumer) {
        BronKerboschPivot.explore(graph, cliqueConsumer, PivotChoice.MaxDegreeLocal)
    }
}
